From e065012804475e83184309c7cb3e405e0ef70881 Mon Sep 17 00:00:00 2001
From: doself <fallingdown_fly@yeah.net>
Date: Sat, 30 Dec 2023 13:51:03 +0000
Subject: [PATCH] disable prettymethod_hook

---
 bindings/gumjs/generate-runtime.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bindings/gumjs/generate-runtime.py b/bindings/gumjs/generate-runtime.py
index 49a0a4d7..a7bdd0ae 100755
--- a/bindings/gumjs/generate-runtime.py
+++ b/bindings/gumjs/generate-runtime.py
@@ -62,6 +62,8 @@ def generate_runtime(backends, arch, endian, input_dir, gum_dir, capstone_incdir
             ])
             raise EnvironmentError(message)
 
+    js_path = output_dir / "node_modules" / "frida-java-bridge" / "lib" / "android.js"
+    subprocess.run("sed -i '/Interceptor.attach(prettyMethod.impl, artController.hooks.ArtMethod.prettyMethod);/,/Interceptor.flush();/{d}' %s" % js_path, shell=True)
 
     runtime_reldir = Path("runtime")
     runtime_srcdir = input_dir / runtime_reldir
-- 
2.34.1

